File: hyein/week2/소수만들기.py
from itertools import combinations
import math

def solution(nums):
    answer = 0
    comb = list(combinations(nums, 3))
    
    for c in comb:
        if is_prime(sum(c)):
            answer += 1

    return answer

def is_prime(num):
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
          
    return True

File: sujeong/week2/소수 만들기.swift
import Foundation

func isPrime(_ num: Int) -> Bool {
    for d in 2..<num {
        if num % d == 0 { return false }
    }
    return true
}

func solution(_ nums:[Int]) -> Int {
    var answer = 0

    for i in 0...nums.count - 3 {
        for ii in i+1...nums.count - 2 {
            for iii in ii+1...nums.count - 1 {
                let sum: Int = nums[i] + nums[ii] + nums[iii]
                if isPrime(sum) { answer += 1 }
            }
        }
    }

    return answer

}
